From ef365c85ab1165876b27a86c37e3462ed4cb81a4 Mon Sep 17 00:00:00 2001
From: YangYonggang <b31664@freescale.com>
Date: Wed, 2 May 2012 18:43:36 +0800
Subject: [PATCH 271/635] ENGR00179835 Mipi 5M size picture color is too dark.

Removed the waiting time after setting the configure in kernel. So need to
discard some frames when change to 5M and back to VGA.

Signed-off-by: YangYonggang <b31664@freescale.com>
---
 mx6/libcamera/CameraHal.cpp     | 9 ++++++++-
 mx6/libcamera/V4l2CsiDevice.cpp | 2 +-
 2 files changed, 9 insertions(+), 2 deletions(-)

diff --git a/mx6/libcamera/CameraHal.cpp b/mx6/libcamera/CameraHal.cpp
index 4a3560a..d1e01f7 100755
--- a/mx6/libcamera/CameraHal.cpp
+++ b/mx6/libcamera/CameraHal.cpp
@@ -1158,7 +1158,6 @@ namespace android {
                 ret = UNKNOWN_ERROR;
                 goto Pic_out;
             }
-
             if (++i == mCaptureDeviceCfg.picture_waite_number)
                 break;
 
@@ -1878,6 +1877,14 @@ Pic_out:
             return INVALID_OPERATION;
         
         isCaptureBufsAllocated = 1;
+
+        unsigned int bufIndex = 0;
+        //skip 10 frames when doing preview
+        for (int k = 0; k < 10; k++) {
+            mCaptureDevice->DevDequeue(&bufIndex);
+            mCaptureDevice->DevQueue(bufIndex);
+        }
+
         for(unsigned int i=0; i < mCaptureBufNum; i++) {
             mCaptureThreadQueue.postMessage(new CMessage(CMESSAGE_TYPE_NORMAL, i));
         }
diff --git a/mx6/libcamera/V4l2CsiDevice.cpp b/mx6/libcamera/V4l2CsiDevice.cpp
index 587dce8..9e09367 100755
--- a/mx6/libcamera/V4l2CsiDevice.cpp
+++ b/mx6/libcamera/V4l2CsiDevice.cpp
@@ -343,7 +343,7 @@ namespace android{
                 pTimeFrame->denominator = 15;
             }
             else if (capturewidth == 2592 && captureheight == 1944) {
-                pic_waite_buf_num =1;
+                pic_waite_buf_num =3;
                 capturemode = 6;	/* 2592x1944 mode */
                 pTimeFrame->numerator = 1;
                 pTimeFrame->denominator = 15;
-- 
1.8.0

